transport.go

```diff
@@ -24,19 +40,27 @@ type Conn interface {
 // but many more can be implemented, sctp, audio signals, sneakernet, UDT, a
 // network of drones carrying usb flash drives, and so on.
 type Transport interface {
-	Dialer(laddr ma.Multiaddr, opts ...DialOpt) (Dialer, error)
```

Comment:
We no longer have "dialers". Instead, we just call `Dial` on the transport and the transport decides how to dial (e.g., which source address to use). In the past, we constructed one dialer per listen address and picked the dialer randomly. That was obviously *not* an optimal solution but the swarm just didn't have enough information to make an informed decision (without special-casing transports). On the other hand, the transport *does* have enough information.